plugins/basic/nexus-repository-raw/src/main/java/org/sonatype/nexus/repository/raw/internal/RawContentFacetImpl.java

```diff
@@ -217,9 +218,12 @@ public boolean delete(final String path) throws IOException {
       if (component == null) {
         return false;
       }
+
+      final SearchFacet searchFacet = getRepository().facet(SearchFacet.class);
+      final SearchItemId searchId = searchFacet.identifier(component);
```

Comment:
I'm coming up short. I've trimmed the 'getRepository()' to lighten it up a tiny bit. Almost makes me wish for:
    tx.addPostCommit(facet(SearchFacet.class).callableDeletion(component) /* Callable */);